library/src/main/java/com/queue_it/androidsdk/QueueActivity.java

```diff
@@ -182,8 +184,45 @@ private void broadcastQueuePassed(String queueItToken) {
         LocalBroadcastManager.getInstance(QueueActivity.this).sendBroadcast(intent);
     }
 
+    private void broadcastQueueAborted() {
+        Intent intent = new Intent("on-queue-aborted");
+        LocalBroadcastManager.getInstance(QueueActivity.this).sendBroadcast(intent);
+    }
+
     private void disposeWebview(WebView webView) {
         webView.loadUrl("about:blank");
         finish();
     }
+
+    @Override
+    public void onBackPressed() {
+//        This is for disable the back button to avoid users killing the Queue activity and jumping over the queue.
+        Log.d("TAG", "Pressed back button");
+        if (!disableExit) {
+            super.onBackPressed();
+        } else {
+            AlertDialog.Builder builder = new AlertDialog.Builder(this);
+            builder.setTitle("Exiting")
+                    .setMessage(R.string.queue_it_exiting)
+                    .setNegativeButton(R.string.cancel, new DialogInterface.OnClickListener() {
+                        @Override
+                        public void onClick(DialogInterface dialog, int which) {
+
+                        }
+                    })
+                    .setPositiveButton(R.string.ok, new DialogInterface.OnClickListener() {
+                        @Override
+                        public void onClick(DialogInterface dialog, int which) {
+                            broadcastQueueAborted();
+                            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN) {
+                                finishAffinity();
```

Comment:
I don't feel that we should exit the app. We should just return to the app.

Reply:
That is also a good idea, just return to the app and clear the cache.